packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -714,7 +715,10 @@ abstract class ResidentRunner {
   ///
   /// Throws an [AssertionError] if [Devce.supportsScreenshot] is not true.
   Future<void> screenshot(FlutterDevice device) async {
-    assert(device.device.supportsScreenshot);
+    if (!device.device.supportsScreenshot) {
+      throwToolExit('Device ${device.device.name} does not support screenshots.');
```

Comment:
Why is this a tool exit, and not just a printError() ?

Reply:
Fixed